src/libsyntax/ast.rs

```diff
@@ -903,6 +917,129 @@ pub struct Expr {
     pub attrs: ThinVec<Attribute>
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ExprPrecedence {
```

Comment:
Could you move all this stuff from `ast.rs` somewhere, e.g. into `util/parser.rs` where it was previously.
This is not a part of AST or HIR, really, it's a parsing-specific detail.

Reply:
Done. Waiting for the tests to run to make sure nothing broke.